### PR TITLE
Support Trunk

### LINKED ISSUE #128: Support Trunk
Hi.

[Trunk](https://github.com/thedodd/trunk) is a popular tool used in GitHub action. If you are OK with this. I will make a PR.

Thanks!

### PR DESCRIPTION
**Pull request by azzamsa**
Fixes #128 

@taiki-e I don't have a clue on where to get the `manifest/trunk.json`

### CODE REVIEW

**Comment by NobodyXu** [comment]
I think it's created using [tools/manifest.sh](https://github.com/taiki-e/install-action/blob/main/tools/manifest.sh)

File: tools/codegen/base/trunk.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "repository": "https://github.com/thedodd/trunk",
+  "tag_prefix": "",
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:3]:
```suggestion
  "tag_prefix": "v",
```

File: tools/codegen/base/trunk.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "asset_name": "${package}-${rust_target}.tar.gz",
+  "platform": {
+    "x86_64_linux_musl": {},
+    "x86_64_macos": {},
+    "x86_64_windows": {
+      "asset_name": "${package}--${rust_target}.zip"
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:9]:
```suggestion
      "asset_name": "${package}-${rust_target}.zip"
```

File: tools/codegen/base/trunk.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "repository": "https://github.com/thedodd/trunk",
+  "tag_prefix": "",
+  "asset_name": "${package}-${rust_target}.tar.gz",
+  "platform": {
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:5]:
```suggestion
  "version_range": "latest",
  "platform": {
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

**Comment by azzamsa** [comment]
it stuck here.

```
⬢ ❯ ./tools/manifest.sh trunk 0.16.0
    Finished release [optimized] target(s) in 0.04s
     Running `target/release/install-action-internal-codegen trunk 0.16.0`
downloading releases of https://github.com/thedodd/trunk from https://api.github.com/repos/thedodd/trunk/releases
update manifest for versions '=0.16.0'
no asset '["trunk-x86_64-unknown-linux-musl.tar.gz"]' for host platform 'x86_64_linux_musl'
downloading https://github.com/thedodd/trunk/releases/download/v0.16.0/trunk-x86_64-apple-darwin.tar.gz for checksum...
```

File: tools/codegen/base/trunk.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "repository": "https://github.com/thedodd/trunk",
+  "tag_prefix": "v",
+  "asset_name": "${package}-${rust_target}.tar.gz",
+  "version_range": "latest",
+  "platform": {
+    "x86_64_linux_musl": {},
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:7]:
```suggestion
```

trunk only has pre-built artifacts for x86-64

File: tools/codegen/base/trunk.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    "x86_64_macos": {},
+    "x86_64_windows": {
+      "asset_name": "${package}-${rust_target}.zip"
+    },
+    "aarch64_linux_musl": {},
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:12]:
```suggestion
```

same

File: tools/codegen/base/trunk.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    "aarch64_macos": {}
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:13]:
```suggestion
```

and it also does not have pre-built artifacts for M1/M2

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: tools/codegen/base/trunk.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "repository": "https://github.com/thedodd/trunk",
+  "tag_prefix": "v",
+  "asset_name": "${package}-${rust_target}.tar.gz",
+  "version_range": "latest",
+  "platform": {
```
Comment (NobodyXu) [inline tools/codegen/base/trunk.json:6]:
```suggestion
  "platform": {
    "x86_64_linux_gnu": {},
```

**Comment by NobodyXu** [comment]
@azzamsa trunks pre-built seems to use glibc 2.19

**Comment by NobodyXu** [comment]
Alpine failed because trunk does not provide x86_64-unknown-linux-musl build.

**Comment by azzamsa** [comment]
> I think the best way to resolve this is for trunk to provide pre-built artifacts for musl and for install-action to also use musl artifacts on gnu targets.

You mean, we need to convince  (or make a PR) to Trunk to include musl target in the release?

**Comment by azzamsa** [comment]
I don't think I can make Trunk compile in musl target.

**Comment by taiki-e** [comment]
musl support is not required...
See [DEVELOPMENT.md](https://github.com/taiki-e/install-action/blob/main/DEVELOPMENT.md#troubleshooting) for how to deal with such an error.

**Comment by azzamsa** [comment]
Something like this #795

**Pull request closed without merging**